Obsidian/WorldData/Generators/Overworld/Decorators/OverworldDecorator.cs

```diff
@@ -1,11 +1,53 @@
 ﻿using Obsidian.ChunkData;
 using Obsidian.WorldData.Generators.Overworld.Features.Flora;
 using Obsidian.WorldData.Generators.Overworld.Features.Trees;
+using System.Diagnostics;
+using System.Linq.Expressions;
 
 namespace Obsidian.WorldData.Generators.Overworld.Decorators;
 
 public static class OverworldDecorator
 {
+    private static readonly ConcurrentDictionary<Type, Func<GenHelper, Chunk, BaseFlora>> floraCache = new();
+    private static readonly ConcurrentDictionary<Type, Func<GenHelper, Chunk, BaseTree>> treeCache = new();
+
+    private static readonly Type[] argumentCache = new[] { typeof(GenHelper), typeof(Chunk) };
+    public static readonly ParameterExpression[] expressionParameters = argumentCache.Select((t, i) => Expression.Parameter(t, $"param{i}")).ToArray();
+
+    static OverworldDecorator()
+    {
+        var asm = typeof(OverworldDecorator).Assembly;
+
+        var floras = asm.GetTypes()
+            .Where(t => t.IsClass && !t.IsAbstract && (t.IsSubclassOf(typeof(BaseFlora)) || t.IsSubclassOf(typeof(BaseTallFlora))));
```

Comment:
BaseTallFlora is a subclass of BaseFlora, might not need the or

Reply:
👀 you right I fix